lumibot/components/agents/managed_gateway.py

```
@@ -0,0 +1,259 @@
+    try:
+        with urllib.request.urlopen(request, timeout=120) as response:
+            return response.status, json.loads(response.read().decode("utf-8"))
+    except urllib.error.HTTPError as exc:
+        try:
+            body = json.loads(exc.read().decode("utf-8"))
+        except Exception:
+            body = {}
+        return exc.code, body
+
+
+class BotSpotManagedLlm(BaseLlm):
```

Pylint (4.0.6)

[refactor] Too few public methods (1/2)

(R0903)